src/Kiota.Builder/Refiners/GoRefiner.cs

```diff
@@ -80,7 +80,7 @@ public override Task Refine(CodeNamespace generatedCode, CancellationToken cance
                     CodePropertyKind.AdditionalData,
                     CodePropertyKind.Custom,
                     CodePropertyKind.BackingStore },
-                static s => s.ToPascalCase(new[] { '_' }),
```

Comment:
PascalCase resulted in collisions when generating the sdk, will move this to a flag as to ensure existing sdks that treat properties such as `query_string` and `queryString` differently still compile

Reply:
What about adding a deduplication logic in the refiner instead?